hphp/runtime/vm/jit/unwind-itanium.cpp

```diff
@@ -358,7 +369,29 @@ void write_tc_cie(EHFrameWriter& ehfw) {
 
   // The following calculation is related to the "top" of the record.  There is
   // an implicit -8 factor as defined by EHFrameWriter::m_cie.data_align.
-  ehfw.offset_extended_sf(dw_reg::IP, (record_size - AROFF(m_savedRip)) / 8);
+
+  switch (arch()) {
+    case Arch::ARM:
+    case Arch::X64:
+      ehfw.offset_extended_sf(dw_reg::IP, (record_size - AROFF(m_savedRip)) / 8);
+      break;
+    case Arch::PPC64:
+      // On PPC64 the return addres of a current frame is found in the parent,
+      // so to get it, the content of the FP need to be taken to achieve the
+      // parent.
```

Comment:
This sentence doesn't make sense syntactically to me.  How about:

`On PPC64, the return address for the current frame is found in the parent frame.  The following expression uses the FP to get the parent frame and recovers the return address from it.`